src/client/pages/CountryHome/Repository/CreateFile/Panel/hooks/useOnSaveFile.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback } from 'react'
+
+import axios from 'axios'
+
+import { ApiEndPoint } from 'meta/api/endpoint'
+
+import { useCountryRouteParams } from 'client/hooks/useRouteParams'
+import { NewFile } from 'client/pages/CountryHome/Repository/CreateFile/Panel/newFile'
+
+type Returned = () => Promise<void>
+
+export const useOnSaveFile = (file: NewFile | null): Returned => {
+  const { assessmentName, cycleName, countryIso } = useCountryRouteParams()
+
+  return useCallback<Returned>(async () => {
+    const formData = new FormData()
+    formData.append('name', file?.name || '')
+    formData.append('link', file?.link || '')
+
+    if (file?.file) {
+      formData.append('file', file.file)
+    }
+
+    await axios.post(ApiEndPoint.CycleData.Repository.one(), formData, {
+      params: {
+        countryIso,
+        assessmentName,
+        cycleName,
+      },
+    })
```

Comment:
@minotogna what do you suggest for handling this state/store/action?
Should we use redux store? (Preferable I guess?)

Currently the Repository state is handled automagically by

```
        <TablePaginated
          header={false}
          counter={false}
          columns={columns}
          path={ApiEndPoint.CycleData.Repository.files()}
        />
```

so we don't have a need for a separate store?

Let me know